src/main/scala/lms/transformation/distributed_tensor/distributedTensorUnary.scala

```diff
@@ -45,6 +69,33 @@ trait FixedSizeDistributedTensorUnaryTypeLess extends FixedSizeDistributedTensor
         (() => {
           Accumulate(gradMap(a), Negate(gradMap(s), anno), anno); ()
         }) +=: backwardNodes
+     
+    case Node(s, "tensor_invert", tt::Backend.Const(anno:Anno)::(a:Backend.Sym)::_, _) =>
+        implicit val pos = Adapter.oldSourceMap(s)
+        forwardNodes += node
+
+        (() => {
+          val a_tensor = new TENSOR(transform(a))
+          val square = Mul(a_tensor, a_tensor, anno)
+          val grad = Div(gradMap(a), square, anno)
+          Accumulate(gradMap(a), grad, anno); ()
+        }) +=: backwardNodes
+    
+    case Node(s, "tensor_tanh", tt::Backend.Const(anno:Anno)::(a:Backend.Sym)::_, _) =>
+        implicit val pos = Adapter.oldSourceMap(s)
+        forwardNodes += node
+
+        (() => {
+          Accumulate(gradMap(a), TanhGrad(gradMap(s), new TENSOR(transform(a)), anno), anno); ()
```

Comment:
TanhGrad might want to take `transform(s)` as input instead? 
otherwise it has to compute tanh again IIRC.

Reply:
done.